build/lib/src/builder/build_step.dart

```diff
@@ -26,19 +25,29 @@ abstract class BuildStep implements AssetReader, AssetWriter {
   @override
   Future<bool> hasInput(AssetId id);
 
+  /// Reads an [Asset] by [id] as a [List<int>].
+  @override
+  Future<List<int>> readAsBytes(AssetId id);
```

Comment:
I should have noticed this earlier... but repeating these signatures here isn't adding any value. The `implements` statement is all we need here. You can move these doc comments to the other classes if they aren't already there.